mayastor/src/bdev/nexus/nexus_share.rs

```diff
@@ -38,26 +40,17 @@ fn validate_frontend_protocol(share_protocol : ShareProtocol) -> Result<SharePro
 }
 
 impl Nexus {
-    /// Publish the nexus to system using nbd device and return the path to
-    /// nbd device.
     pub async fn share(
         &mut self,
         share_proto: ShareProtocol,
         key: Option<String>,
     ) -> Result<String, Error> {
-        if self.nbd_disk.is_some() {
-            return Err(Error::AlreadyShared {
-                name: self.name.clone(),
-            });
-        }
 
         assert_eq!(self.share_handle, None);
         validate_frontend_protocol(share_proto)?;
 
         self.share_protocol = share_proto;
```

Comment:
Should not do this make already be shared, possibly with different protocol.

Reply:
Ok. Also need to check that *both* nbd_disk and iscsi_target are None in both cases.